nornir/core/deserializer/configuration.py

```diff
@@ -0,0 +1,184 @@
+        description="Import path to inventory plugin",
+    )
+    options: Dict[str, Any] = Schema(
+        default={}, description="kwargs to pass to the inventory plugin"
+    )
+    transform_function: Any = Schema(
+        default="",
+        description=(
+            "Path to transform function. The transform_function "
+            "you provide will run against each host in the inventory"
+        ),
+    )
+
+    class Config:
+        env_prefix = "NORNIR_INVENTORY_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(self, **kwargs) -> configuration.InventoryConfig:
+        inv = InventoryConfig(**kwargs)
+        return configuration.InventoryConfig(
+            plugin=cast(Type[Inventory], _resolve_import_from_string(inv.plugin)),
+            options=inv.options,
+            transform_function=_resolve_import_from_string(inv.transform_function),
+        )
+
+
+class LoggingConfig(BaseSettings):
+    level: str = Schema(default="debug", description="Logging level")
+    file: str = Schema(default="nornir.log", descritpion="Logging file")
+    format: str = Schema(
+        default="%(asctime)s - %(name)12s - %(levelname)8s - %(funcName)10s() - %(message)s",
+        description="Logging format",
+    )
+    to_console: bool = Schema(
+        default=False, description="Whether to log to console or not"
+    )
+    loggers: List[str] = Schema(default=["nornir"], description="Loggers to configure")
+
+    class Config:
+        env_prefix = "NORNIR_LOGGING_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(self, **kwargs) -> configuration.LoggingConfig:
+        logging_config = LoggingConfig(**kwargs)
+        return configuration.LoggingConfig(
+            level=getattr(logging, logging_config.level.upper()),
+            file_=logging_config.file,
+            format_=logging_config.format,
+            to_console=logging_config.to_console,
+            loggers=logging_config.loggers,
+        )
+
+
+class Jinja2Config(BaseSettings):
+    filters: str = Schema(
+        default="", description="Path to callable returning jinja filters to be used"
+    )
+
+    class Config:
+        env_prefix = "NORNIR_JINJA2_"
+        ignore_extra = False
+
+    @classmethod
+    def deserialize(self, **kwargs) -> configuration.Jinja2Config:
+        c = Jinja2Config(**kwargs)
+        jinja_filter_func = _resolve_import_from_string(c.filters)
+        jinja_filters = jinja_filter_func() if jinja_filter_func else {}
+        return configuration.Jinja2Config(filters=jinja_filters)
+
+
+class CoreConfig(BaseSettings):
+    num_workers: int = Schema(
+        default=20,
+        description="Number of Nornir worker threads that are run at the same time by default",
+    )
+    raise_on_error: bool = Schema(
+        default=False,
+        description=(
+            "If set to ``True``, (:obj:`nornir.core.Nornir.run`) method of "
+            "will raise an exception if at least a host failed"
```

Comment:
Can we please add a reference to `nornir.core.exceptions.NornirExecutionError` here?